server/startup/migrations/v236.ts

```diff
@@ -0,0 +1,125 @@
+		copySettingValue('LDAP_Update_Data_On_Login', 'LDAP_Sync_User_Data');
+
+		let fieldMap;
+		try {
+			fieldMap = JSON.parse(settings.get<string>('LDAP_Sync_User_Data_FieldMap') ?? '');
+		} catch (_error) {
+			// Ignore any parsing errors;
+		}
+
+		if (fieldMap) {
+			const newObject: Record<string, string> = {};
+
+			for (const key in fieldMap) {
+				if (!fieldMap.hasOwnProperty(key)) {
+					continue;
+				}
+
+				if (fieldMap[key] === 'name') {
+					Settings.upsert({ _id: 'LDAP_Name_Field' }, { $set: { value: key } });
+					Settings.upsert({ _id: 'LDAP_AD_Name_Field' }, { $set: { value: key } });
+					continue;
+				}
+
+				if (fieldMap[key] === 'email') {
+					Settings.upsert({ _id: 'LDAP_Email_Field' }, { $set: { value: key } });
+					Settings.upsert({ _id: 'LDAP_AD_Email_Field' }, { $set: { value: key } });
+					continue;
+				}
+
+				newObject[fieldMap[key]] = key;
+			}
+
+			if (isEE) {
+				const newJson = JSON.stringify(newObject);
+				Settings.upsert({ _id: 'LDAP_CustomFieldMap' }, { $set: { value: newJson } });
+
+				const syncCustomFields = Object.keys(newObject).length > 0 && settings.get('LDAP_Sync_User_Data');
+				Settings.upsert({ _id: 'LDAP_Sync_Custom_Fields' }, { $set: { value: syncCustomFields } });
+			}
+		}
+
+		if (isEE) {
+			copySettingValue('LDAP_Sync_User_Data_Roles', 'LDAP_Sync_User_Data_Groups');
+			copySettingValue('LDAP_Sync_User_Data_Roles_AutoRemove', 'LDAP_Sync_User_Data_Groups_AutoRemove');
+			copySettingValue('LDAP_Sync_User_Data_Roles_Filter', 'LDAP_Sync_User_Data_Groups_Filter');
+			copySettingValue('LDAP_Sync_User_Data_Roles_BaseDN', 'LDAP_Sync_User_Data_Groups_BaseDN');
+			copySettingValue('LDAP_Sync_User_Data_RolesMap', 'LDAP_Sync_User_Data_GroupsMap');
+			copySettingValue('LDAP_Sync_User_Data_Channels', 'LDAP_Sync_User_Data_Groups_AutoChannels');
+			copySettingValue('LDAP_Sync_User_Data_Channels_Admin', 'LDAP_Sync_User_Data_Groups_AutoChannels_Admin');
+			copySettingValue('LDAP_Sync_User_Data_ChannelsMap', 'LDAP_Sync_User_Data_Groups_AutoChannelsMap');
+			copySettingValue('LDAP_Sync_User_Data_Channels_Enforce_AutoChannels', 'LDAP_Sync_User_Data_Groups_Enforce_AutoChannels');
+
+			copySettingValue('LDAP_Sync_User_Data_Channels_Filter', 'LDAP_Sync_User_Data_Groups_Filter');
+			copySettingValue('LDAP_Sync_User_Data_Channels_BaseDN', 'LDAP_Sync_User_Data_Groups_BaseDN');
+		} else {
+			Settings.remove({
+				_id: {
+					$in: [
+						'LDAP_Authentication',
+						'LDAP_Authentication_UserDN',
+						'LDAP_Authentication_Password',
+						'LDAP_Background_Sync',
+						'LDAP_Background_Sync_Interval',
+						'LDAP_Background_Sync_Import_New_Users',
+						'LDAP_Background_Sync_Keep_Existant_Users_Updated',
+					],
+				},
+			});
+		}
+
+		Settings.remove({
+			_id: {
+				$in: [
+					'LDAP_Sync_Now',
+					'LDAP_Test_Connection',
+					'LDAP_Sync_CustomFields',
+					'LDAP_Sync_User_Data',
+					'LDAP_Sync_User_Data_FieldMap',
+					'LDAP_Enable_LDAP_Roles_To_RC_Roles',
```

Comment:
this settings is still used at https://github.com/RocketChat/Rocket.Chat/blob/0a293e547fcab88d548d14b6c8608b530b5d6f9d/app/statistics/server/lib/getServicesStatistics.ts#L43

also, this and some other settings are mentioned on this README file, which I think we can get rid off, right? 
https://github.com/RocketChat/Rocket.Chat/blob/0a293e547fcab88d548d14b6c8608b530b5d6f9d/ee/app/ldap-enterprise/README.md#L18